public/app/features/dashboard/dashgrid/DashboardLibrary/api/dashboardLibraryApi.test.ts

```diff
@@ -80,8 +67,74 @@ describe('dashboardLibraryApi', () => {
   });
 
   describe('fetchCommunityDashboards', () => {
+    describe('filterNotSafeDashboards', () => {
+      it('should filter out dashboards with panel types that can contain JavaScript code', async () => {
+        const safeDashboard = createMockGnetDashboardWithDownloads({ id: 1 });
+        const mockDashboards = [
+          safeDashboard,
+          createMockGnetDashboardWithDownloads({ id: 2, panelTypeSlugs: ['ae3e-plotly-panel'] }),
+        ];
+        const mockResponse = {
+          page: 1,
+          pages: 5,
+          items: mockDashboards,
+        };
+        mockGet.mockResolvedValue(mockResponse);
+
+        const result = await fetchCommunityDashboards(defaultFetchParams);
+
+        expect(result).toEqual({
+          page: 1,
+          pages: 5,
+          items: [safeDashboard],
+        });
+      });
+
+      it('should filter out dashboards with low downloads', async () => {
+        const safeDashboard = createMockGnetDashboardWithDownloads({ id: 1 });
+        const mockDashboards = [safeDashboard, createMockGnetDashboard({ id: 2, downloads: 999 })];
+        const mockResponse = {
+          page: 1,
+          pages: 5,
+          items: mockDashboards,
+        };
+        mockGet.mockResolvedValue(mockResponse);
+
+        const result = await fetchCommunityDashboards(defaultFetchParams);
+
+        expect(result).toEqual({
+          page: 1,
+          pages: 5,
+          items: [safeDashboard],
+        });
+      });
+    });
+
+    it('should not return more than MAX_PAGES_FILTER pages due to security reasons', async () => {
```

Comment:
is this test still relevant if we are not using pagination feature currently? i suppose we are just testing the api so maybe its ok to keep as api supports pagination